colrev/ops/built_in/search_sources/robert_semanticscholar.py

```diff
@@ -0,0 +1,365 @@
+#! /usr/bin/env python
+"""SearchSource: Semantic Scholar"""
+
+from __future__ import annotations
+import json
+import typing
+
+from semanticscholar import PaginatedResults
+
+import semanticscholarui
+import semanticscholarvalidator
+
+from dataclasses import dataclass
+from multiprocessing import Lock
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import requests as requests
+# install zope package
+import zope.interface
+# install package dacite
+from dacite import from_dict
+from dataclasses_jsonschema import JsonSchemaMixin
+
+import colrev.env.package_manager
+import colrev.exceptions as colrev_exceptions
+import colrev.ops.built_in.search_sources.doi_org as doi_connector
+import colrev.ops.built_in.search_sources.utils as connector_utils
+import colrev.record
+import colrev.settings
+import colrev.operation
+import colrev.env.language_service
+import colrev.review_manager
+import colrev.env.environment_manager
+from colrev import settings
+from colrev.constants import Colors
+from colrev.constants import Fields
+from colrev.constants import FieldValues
+
+from semanticscholar import SemanticScholar
+
+if TYPE_CHECKING:
+    import colrev.ops.search
+    import colrev.ops.prep
+
+
+# pylint: disable=unused-argument
+# pylint: disable=duplicate-code
+# pylint: disable=too-many-lines
+
+@zope.interface.implementer(
+    colrev.env.package_manager.SearchSourcePackageEndpointInterface
```

Comment:
As a quick note: The interface implementation annotation means that the SemanticScholarSearchSource should implement (define) all attributes and methods required by the SearchSourcePackageEndpointInterface, as stated in the [documentation](https://colrev.readthedocs.io/en/latest/dev_docs/extensions.html#colrev.env.package_manager.SearchSourcePackageEndpointInterface). Some of the methods, such as load, prep, get_masterdata are missing, which explains the zope.interface.exceptions.MultipleInvalid error. As a starting point, you could simply define empty methods.